docs/developers/code-style.md

```diff
@@ -220,6 +365,162 @@ where:
+    // then
+    assertThat(foo).isEqualTo("foo");
+}
+
+@Test
+public void testBar() {
+    // when
+    final String bar = service.getBar();
+    
+    // then
+    assertThat(bar).isEqualTo("bar");
+}
+```
+
+This also applies to cases where same method is tested with different arguments inside single unit test.
+Note: This represents the replacement we have selected for parameterized testing.
+
+```
+// bad
+@Test
+public void testFooFirstSecond() {
+    // when
+    final String foo1 = service.getFoo(1);
+    final String foo2 = service.getFoo(2);
+    
+    // then
+    assertThat(foo1).isEqualTo("foo1");
+    assertThat(foo2).isEqualTo("foo2");
+}
+
+// good
+@Test
+public void testFooFirst() {
+    // when
+    final String foo1 = service.getFoo(1);
+    
+    // then
+    assertThat(foo1).isEqualTo("foo1");
+}
+
+@Test
+public void testFooSecond() {
+    // when
+    final String foo2 = service.getFoo(2);
+    
+    // then
+    assertThat(foo2).isEqualTo("foo2");
+}
+```
+
+#### Unit tests naming
+
+Name unit tests meaningfully. Test names should give brief description of what unit test tries to check.
+It is also recommended to prefix test methods with the name of method that is being tested and word `should`.
+
+```
+// bad
+@Test
+public void doSomethingTest() {
+    // when and then
+    assertThat(service.processData()).isEqualTo("result");
+}
+
+// good
+@Test
+public void processDataShouldReturnResult() {
+    // when and then
+    assertThat(service.processData()).isEqualTo("result");
+}
+```
+
+#### Place test data as close as possible to test
+
+Place data used in test as close as possible to test code. This will make tests easier to read, review and understand.
+
+```
+// bad
+@Test
+public void testFoo() {
+    // given
+    final String fooData = getSpecificFooData();
+    
+    // when and then
+    assertThat(service.processFoo(fooData)).isEqualTo(getSpecificFooResult());
+}
+
+// good
+@Test
+public void testFoo() {
+    // given
+    final String fooData = "fooData";
+    
+    // when and then
+    assertThat(service.processFoo(fooData)).isEqualTo("fooResult");
+}
+```
+
+This point also implies the next one.
+
+#### Avoid class level constants in test classes
+
+Since we are trying to improve test simplicity and readability and place test data close to tests, we decided to avoid usage of top level constants where it is possible.
+Instead, just inline constant values.
+
+```
+// bad
+public class TestClass {
+
+    private static final String CONSTANT_1 = "foo";
+    ...
+    private static final String CONSTANT_N = "bar";
+    
+    // A bunch of other tests
+    
+    @Test
+    public void testFoo() {
+        // when and then
+        assertThat(service.foo(CONSTANT_1)).isEqualTo(CONSTANT_N);
+    }
+}
+
+// good
+public class TestClass {
+    // A bunch of other tests
+    
+    @Test
+    public void testFoo() {
+        // when and then
+        assertThat(service.foo("foo")).isEqualTo("bar");
+    }
+}
+```
```

Comment:
I prefer using constant especially if it's used a few times especially in separate tests, constants doesn't decrease the readability at all if they have good names